content/concepts/decentralized-exchange/automated-market-makers.md

```diff
@@ -0,0 +1,88 @@
+
+- Liquidity providers can redeem their LP Tokens to withdraw a corresponding share of the assets in the AMM's pool, including fees collected.
+- Liquidity providers can vote to change the AMM's fee settings. The votes are weighted based on how many LP Tokens the voters hold.
+- Liquidity providers can bid some of their LP Tokens to receive a temporary discount on the AMM's trading fees.
+
+When the flow of funds between the two assets in a pool is relatively active and balanced, the fees provide a source of passive income for liquidity providers. However, when the relative price between the assets shifts, the liquidity providers can take a loss on the [currency risk](https://www.investopedia.com/terms/c/currencyrisk.asp).
+
+## How the AMM Works
+
+[Kris Machowski's Introduction to Automated Market Makers](https://www.machow.ski/posts/an_introduction_to_automated_market_makers/) is a good in-depth explanation on the economics and principles of AMMs in general. The XRP Ledger implements a _constant product_ AMM, which sets the exchange rate between the two assets in its pool so that multiplying the amounts of the two pools together always comes out to the same value. For example, if the AMM holds 5 ETH and 5 USD, the "constant product" value is 5×5=25. To buy 1 ETH, you'd have to spend 1.25 USD, because after removing 1 ETH from the pool and adding 1.25 USD, the constant product value is 4×6.25=25. The AMM also charges a configurable trading fee, which can be as low as 0 or as high as 1%, so you would have to pay a little more for that as well.
+
+**Note:** The XRP Ledger's AMM is designed to be extensible so that, with future protocol amendments, it would be possible to create an AMM where the pools are be weighted unevenly, which would make it a more general _geometric mean market maker_. However, the current implementation always uses a weight parameter of 0.5, so it functions like a constant product AMM.
+
+Due to the way the constant product rule works, the bigger a proportion of the AMM's pools a trade involves, the higher the costs. This also means that, the larger the amounts in the AMM's pools, the lower the cost of making a specific trade. For example, trading 5 USD through an AMM that only holds 10 USD is more expensive than trading 5 USD through an AMM that holds 1000 USD. This is separate from the trading fee, which is always strictly a percentage of the amounts involved.
+
+An AMM can hold two assets: at most one of these can be XRP, and one or both of them can be [tokens](tokens.html). Tokens with different issuers are considered different assets for this purpose. This means that there can be an AMM for two tokens with the same currency code but different issuers ("FOO issued by WayGate" is different than "FOO issued by StableFoo"), or the same issuer but different currency codes. The order does not matter; the AMM for FOO.WayGate to XRP is the same as for XRP to FOO.WayGate.
```

Comment:
I'd add this as a note above, when we talk about creating AMMs from unique asset pairs.

Reply:
Moved this closer to the beginning of the "How the AMM Works" section